components/Typography/src/UIFont+SimpleEquality.h

```diff
@@ -0,0 +1,33 @@
+/*
+ Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+/**
+ The Typography component provides methods for getting sized fonts and opacities following Material
+ style guidelines.
+
+ This header is the umbrella header for the component and should be imported by consumers of the
+ Typography component. Please do not directly import other headers. This will allow the componet to
+ expand or contract the header file space without consumer modifications.
+ */
+
+#import <UIKit/UIKit.h>
+
+@interface UIFont (SimpleEquality)
+
+/* Checks simple characteristics: name, weight, pointsize. */
+- (BOOL)isEqualToFont:(UIFont*)font;
```

Comment:
Because this is an extension it will require an mdc_ prefix.

---

Alternatively — because it's possible we might re-implement this same extension elsewhere — it may be worth considering making this a C API. This will de-risk the possibility of accidentally introducing this extension in multiple places within MDC in the future and having to deal with very difficult-to-debug behavior.

For context, duplicate Objective-C extension APIs will build successfully with no warning (though if this has changed recently that would be wonderful), resulting in one of the implementations being chosen (essentially at random) at runtime. C APIs on the other hand will not pass the linker phase if multiple APIs exist with the same symbol.

So the end result would look roughly like so:

```objc
FOUNDATION_EXPORT
BOOL MDCFontIsEqualToFont(UIFont *font1, UIFont *font2);
```

Reply:
Done.